docs/gateway-api.md

```diff
@@ -0,0 +1,90 @@
+## Gateway API
+
+Gateway API is L4 and L7 layer routing project in Kubernetes. It represents next generation of k8s Ingress, LB and Service Mesh APIs. For more information on the project see: [Gateway API SIG.](https://gateway-api.sigs.k8s.io/)
+
+**Move from Ingress to Gateway APIs**
+Since Gateway APIs are successor to Ingress Controllers there needs to be a one time migration from Ingress -> GW API resources. To learn more about it refer to: [Ingress Migration](https://gateway-api.sigs.k8s.io/guides/migrating-from-ingress/#migrating-from-ingress)
+
+
+### Resource Models in Gateway API
+There are 3 main resource models in gateway apis:
+1. GatewayClass - Mostly managed by a controller.
+2. Gateway - An instance of traffic handling infra like a LB.
+3. Routes - Defines HTTP-specific rules for mapping traffic from a Gateway listener to a representation of backend network endpoints.
+
+**k8s Gateway API is NOT the same as API Gateways**
+While both sound the same, API Gateway is a more of a general concept that defines a set of resources that exposes capabilities of a backend service but also provide other functionalities like traffic management, rate limiting, authentication and more. It is geared towards commercial API management and monetisation. 
+
+From the gateway api sig:
+> [!NOTE]
```

Comment:
our docs now support Admonitions, we can now use markdown for these types of notes.

```suggestion
!!! note
   Most Gateway API implementations are API Gateways to some extent, but not all API Gateways are Gateway API implementations.
```